lib/matplotlib/tests/test_backend_svg.py

```diff
@@ -150,48 +152,78 @@ def _test_determinism_save(filename, usetex):
     FigureCanvasSVG(fig).print_svg(filename)
 
 
-def _test_determinism(filename, usetex):
-    import os
-    import sys
-    from subprocess import check_output, STDOUT, CalledProcessError
-    from nose.tools import assert_equal
-    plots = []
+def _test_determinism_helper(pipe, usetex):
+    guard = list({str(i): i for i in range(10)})
```

Comment:
Running this line, I get what it does, but it's going to need a comment lest someone "optimize" it into a more straightforward list comprehension.